sdcm/scan_operation_thread.py

```diff
@@ -307,28 +296,81 @@ def execute_query(self, session, cmd: str):
+
+        with tempfile.NamedTemporaryFile(mode='w+', delete=True, encoding='utf-8') as reorder_normal_query_output:
+            cmd = f'tac {self.normal_query_output.name}'
+            if self.limit:
+                cmd += f' | head -n {self.limit}'
+            cmd += f' > {reorder_normal_query_output.name}'
+            self._run_shell_command(cmd=cmd)
+            reorder_normal_query_output.flush()
+            self._run_shell_command(cmd=f'cp {reorder_normal_query_output.name} {self.normal_query_output.name}')
+            self.normal_query_output.flush()
+
+        file_names = f' {self.normal_query_output.name} {self.reversed_query_output.name}'
+        diff_cmd = f"diff -y --suppress-common-lines {file_names}"
+        self.log.info("Comparing scan queries output files by: %s", diff_cmd)
+        stdout, stderr = self._run_shell_command(cmd=diff_cmd)
+        if not stderr:
+            if not stdout:
+                self.log.info("Compared output of normal and reversed queries is identical!")
+            else:
+                self.log.warning("Normal and reversed queries output differs: \n%s", stdout.strip())
+                ls_cmd = f"ls -alh {file_names}"
+                head_cmd = f"head {file_names}"
+                for cmd in [ls_cmd, head_cmd]:
+                    stdout, stderr = self._run_shell_command(cmd=cmd)
+                    if stdout:
+                        stdout = stdout.strip()
+                    self.log.info("%s command output is: \n%s", cmd, stdout)
+        self.reset_output_files()
+
+    def run_scan_operation(self, cmd: str = None, update_stats: bool = True):  # pylint: disable=too-many-locals
         queries = self.randomly_form_cql_statement()
         if not queries:
             return
         normal_query, reversed_query = queries
-        ScanOperationThread.run_scan_operation(self, scan_operation_event=scan_operation_event, cmd=reversed_query)
+        self.scan_event = FullPartitionScanReversedOrderEvent
+        ScanOperationThread.run_scan_operation(self, cmd=reversed_query)
         self.reversed_query_filter_ck_stats[self.ck_filter]['count'] += 1
         self.reversed_query_filter_ck_stats[self.ck_filter]['total_scan_duration'] += self.time_elapsed
         count = self.reversed_query_filter_ck_stats[self.ck_filter]['count']
         average = self.reversed_query_filter_ck_stats[self.ck_filter]['total_scan_duration'] / count
         self.log.info('Average %s scans duration of %s executions is: %s', self.ck_filter, count, average)
         if self.full_partition_scan_params.get('validate_data'):
-            # TODO: implement when a new hydra docker with deepdiff is available
-            self.log.debug('Temporarily not executing the normal query of: %s', normal_query)
+            self.log.debug('Executing the normal query: %s', normal_query)
+            self.scan_event = FullPartitionScanEvent
+            ScanOperationThread.run_scan_operation(self, cmd=normal_query)
```

Comment:
why are you running again `run_scan_operation` here?
this `if` shouldn't go before line 353?

Reply:
the reversed-query has to run first, always, in any case.
Then if 'data_validation' is configured, it runs the normal-query to compare results.

Reply:
so it is related to the verification of the data only... got it